### DISCUSSION TITLE
JSON object to query string

### DISCUSSION THREAD

**Discussion by dr3adx** [nlohmann/json#2277] [Q&A] [answered]
Hi guys,

I'm trying to make some API calls to Binance but their server only accepts **www-form-urlencoded** as content type.
So I have to convert a **nested** JSON object to a query string.

Object example:

```
{
    "batchOrders":
 [
   {
        "test": "testString",
        "test1": "testString2"
    },
   {
        "test": "testString",
        "test1": "testString2"
    },
 ]
}

```
So is there any way to transform nested JSON object into a query string?

**Comment by nlohmann** [comment]
What would be the desired result?

**Reply by nlohmann** [reply]
?

**Comment by Matesanz** [comment, marked as answer]
I want to transform **JSON** to **Query String**:

I have:
```
{ one: 'first', two: 'second' }
```
and I want:
```
one=first&two=second
```

How can it be done? Thanks.

**Reply by nlohmann** [reply]
You can iterate key/values of an object with the [`items()`](https://nlohmann.github.io/json/api/basic_json/items/) member function.

**Reply by Matesanz** [reply]
Thanks! :smile:

**Reply by IssaAboudi** [reply]
> You can iterate key/values of an object with the [`items()`](https://nlohmann.github.io/json/api/basic_json/items/) member function.

Hi, this has helped tremendously but I'm a little lost as for how to store the key and the values into variables.

**Reply by nlohmann** [reply]
What about

```cpp
std::string s;

for (auto x : j.items())
{
    s += x.key();
}
```
